nixos/modules/tasks/flake-auto-upgrade.nix

```
@@ -0,0 +1,295 @@
+        (mkIf (cfg.gitCryptKeyFile != null) [ pkgs.git-crypt ])
+        [ pkgs.git ]
+      ];
+      serviceConfig = mkMerge [
+        {
+          Type = "oneshot";
+          CapabilityBoundingSet = "";
+          LockPersonality = true;
+          ProtectHostname = true;
+          ProtectKernelLogs = true;
+          ProtectKernelModules = true;
+          ProtectKernelTunables = true;
+          PrivateDevices = true;
+          WorkingDirectory = "/var/lib/${name}";
+          StateDirectory = name;
+          RestrictNamespaces = true;
+          RestrictRealtime = true;
+          DynamicUser = true;
+        }
+        (mkIf (cfg.credentials != null) {
+          LoadCredential = "passwordFile:${cfg.credentials.passwordFile}";
+        })
+        (mkIf (cfg.ssh != null) { LoadCredential = "sshKey:${cfg.ssh.key}"; })
+      ];
+      script = ''
+        if [ ! -d repo ]; then
+          ${pkgs.git}/bin/git clone ${cfg.remote} repo
+          cd repo || exit
+        else
+          cd repo || exit
+          ${pkgs.git}/bin/git fetch
+        fi
+        ${lib.optionalString (cfg.gitCryptKeyFile != null) ''
+          ${pkgs.git-crypt}/bin/git-crypt unlock ${cfg.gitCryptKeyFile}
+        ''}
+        rm -f ./*.log
+        if ${pkgs.git}/bin/git checkout ${cfg.updateBranch};
+        then
+          # We don't pull because a force push could have happend in the meantime
+          ${pkgs.git}/bin/git reset --hard origin/${cfg.updateBranch} || true
+        else
+          ${pkgs.git}/bin/git checkout -b ${cfg.updateBranch}
+        fi
+        ${pkgs.git}/bin/git merge origin/${cfg.mainBranch} || true
+        ${lib.optionalString cfg.updateFlake ''
+          ${nixCommand} flake update --commit-lock-file
+        ''}
+        ${cfg.updateScript}
+        ${mkBuilds cfg.buildAttributes}
+        ${lib.optionalString cfg.failLogInCommitMsg ''
+          oldmessage=$(${pkgs.git}/bin/git log -n1 --pretty=%B)
+          faillog=$(cat fail.log | ${pkgs.gawk}/bin/awk '{print $0 " failed"}')
+          ${pkgs.git}/bin/git commit --amend -m "$(echo $oldmessage; printf "\n\n"; echo $faillog)"
+        ''}
+        ${cfg.postCommands}
+        ${pkgs.git}/bin/git push || ${pkgs.git}/bin/git push --set-upstream origin ${cfg.updateBranch}
+      '';
+      startAt = cfg.dates;
+    };
+  mkTimers = name: cfg:
+    nameValuePair name {
+      timerConfig = {
+        RandomizedDelaySec = cfg.randomizedDelaySec;
+        Persistent = cfg.persistent;
+      };
+    };
+in {
+  options = {
+
+    services.flakeAutoUpgrade = mkOption {
+
+      description = (mdDoc ''
+        This add an systemd service NAME,
+        which automatically updates an flake and builds it
+      '');
+      default = { };
+      type = types.attrsOf (types.submodule (
+
+        { ... }: {
+          options = {
+            remote = mkOption {
+              type = types.nullOr types.str;
+              description = lib.mdDoc "Remote to fetch from";
```

```suggestion
              description = "Remote to fetch from";
```
Remove all uses of `lib.mdDoc`